crates/store/re_data_loader/src/le_robot/mod.rs

```diff
@@ -0,0 +1,275 @@
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct LeRobotDatasetInfo {
+    pub robot_type: String,
+    pub codebase_version: String,
+    pub total_episodes: usize,
+    pub total_frames: usize,
+    pub total_tasks: usize,
+    pub total_videos: usize,
+    pub total_chunks: usize,
+    pub chunks_size: usize,
+    pub data_path: String,
+    pub video_path: String,
+    pub fps: usize,
+    pub features: HashMap<String, Feature>,
+}
+
+impl LeRobotDatasetInfo {
+    pub fn load_from_file(filepath: impl AsRef<Path>) -> Result<Self, LeRobotError> {
+        let info_file = File::open(filepath)?;
+        let reader = BufReader::new(info_file);
+
+        serde_json::from_reader(reader).map_err(|err| err.into())
+    }
+
+    pub fn feature(&self, feature_key: &str) -> Option<&Feature> {
+        self.features.get(feature_key)
+    }
+
+    pub fn get_chunk_index(&self, episode_index: usize) -> Result<usize, LeRobotError> {
+        if episode_index > self.total_episodes {
+            return Err(LeRobotError::InvalidEpisodeIndex(episode_index));
+        }
+
+        // chunk indices start at 0
+        let chunk_idx = episode_index / self.chunks_size;
+        if chunk_idx < self.total_chunks {
+            Ok(chunk_idx)
+        } else {
+            Err(LeRobotError::InvalidChunkIndex(chunk_idx))
+        }
+    }
+
+    pub fn episode_data_path(&self, episode_index: usize) -> Result<PathBuf, LeRobotError> {
+        let chunk = self.get_chunk_index(episode_index)?;
+
+        // TODO(gijsd): Need a better way to handle this, as this only supports the default.
+        Ok(self
+            .data_path
+            .replace("{episode_chunk:03d}", &format!("{chunk:03}"))
+            .replace("{episode_index:06d}", &format!("{episode_index:06}"))
+            .into())
+    }
+
+    /// Get the path to a video observation for a specific episode index.
+    pub fn video_path(
+        &self,
+        observation_key: &str,
+        episode_index: usize,
+    ) -> Result<PathBuf, LeRobotError> {
+        let chunk = self.get_chunk_index(episode_index)?;
+        let feature = self
+            .feature(observation_key)
+            .ok_or(LeRobotError::InvalidFeatureKey(observation_key.to_owned()))?;
+
+        if feature.dtype != DType::Video {
+            return Err(LeRobotError::InvalidFeatureDtype {
+                key: observation_key.to_owned(),
+                expected: DType::Video,
+                actual: feature.dtype,
+            });
+        }
+
+        // TODO(gijsd): Need a better way to handle this, as this only supports the default.
+        Ok(self
+            .video_path
+            .replace("{episode_chunk:03d}", &format!("{chunk:03}"))
+            .replace("{episode_index:06d}", &format!("{episode_index:06}"))
+            .replace("{video_key}", observation_key)
+            .into())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Feature {
+    pub dtype: DType,
+    pub shape: Vec<usize>,
+}
```

Comment:
Docstrings!

(What is the shape of a bool? O.o)

Reply:
The shape directly correlates to the shape of the pytorch tensor that LeRobot creates when loading the dataset.
So for a bool this would be `[1,]`, for a 224x224 rgb image this would be `[224, 224, 3]`.